nab/detectors/bayes_changept/bayes_changept_detector.py

```diff
@@ -0,0 +1,164 @@
+"""
+Anomaly detection algorithm that implements Bayesian changepoint detection, as
+described in
+  Ryan P. Adams, David J.C. MacKay, Bayesian Online Changepoint Detection,
+arXiv 0710.3742 (2007).
+
+For the author's implementation (in MATLAB) please see
+http://hips.seas.harvard.edu/content/bayesian-online-changepoint-detection
+
+It has been modified from the author's code to run online: rather than
+initializing a run-length matrix with the size of the dataset, we use an array
+that recursively overwrites old data (no longer needed by the algorithm).
+"""
```

Comment:
We should describe here the various parameters that were tuned, and the effort made to get this working well. Given the poor results, this will be a major concern if a machine learning person looks at this code.